client/api/index.js

```diff
@@ -141,8 +141,20 @@ export default class WCStripeAPI {
 		} );
 	}
 
+	/**
+	 * Creates a setup intent without confirming it.
+	 *
+	 * @return {Promise} The final promise for the request to the server.
+	 */
 	initSetupIntent() {
-		console.error( 'TODO: Not implemented yet: initSetupIntent' );
+		return this.request( getAjaxUrl( 'init_setup_intent' ), {
+			_ajax_nonce: getStripeServerData()?.createSetupIntentNonce,
+		} ).then( ( response ) => {
+			if ( ! response.success ) {
+				throw response.data.error;
+			}
+			return response.data;
+		} );
```

Comment:
```suggestion
		} ).then( ( response ) => {
			if ( ! response.success ) {
				throw response.data.error;
			}
			return response.data;
		} ).catch( ( error ) => {
		    	if ( error.message ) {
					throw error;
				} else {
					// Covers the case of error on the Ajax request.
					throw new Error( error.statusText );
				}
		} );
```
This is a minor moan, but it wouldn't hurt to add this `catch` block to the AJAX request here. This was added on other AJAX requests, because if there is a server error (timeout, lost connection, etc.: you can simulate this case by removing a semi-colon or something in the gateway code to force the server to respond with a `500` status), the response will not contain expected fields, such as `success` or `data`; instead the response will be a standard XHR response object. If there is no `error` property on the object, we create an `Error` object with this parameter and throw it up the chain for the calling function to handle it, by showing an error message. Apologies for the long-winded explanation, but hopefully some of that made sense.

Reply:
Good catch 😜